src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/jpa/repository/GroupRepository.kt

```diff
@@ -1,10 +1,37 @@
 package uk.gov.justice.digital.hmpps.externalusersapi.jpa.repository
 
-import org.springframework.data.repository.CrudRepository
+import kotlinx.coroutines.flow.Flow
+import org.springframework.data.r2dbc.repository.Query
+import org.springframework.data.repository.kotlin.CoroutineSortingRepository
+import org.springframework.lang.NonNull
 import uk.gov.justice.digital.hmpps.externalusersapi.model.Group
+import java.util.UUID
 
-interface GroupRepository : CrudRepository<Group, String> {
+interface GroupRepository : CoroutineSortingRepository<Group, String> {
+  fun findAllByOrderByGroupName(): Flow<Group>
 
-  fun findAllByOrderByGroupName(): List<Group>
-  fun findByGroupCode(groupCode: String?): Group?
+  suspend fun findByGroupCode(groupCode: String?): Group?
```

Comment:
Do we need to make sure , return type should be Mono or Flow

Reply:
TODO 22 - already referenced above, but ensure this issue doesn't get missed - so adding a TODO !! Make a team decision on **return type should be Mono or Flow**